extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/transfer/DataPlanePublicApiController.java

```diff
@@ -0,0 +1,113 @@
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.OutputStreamDataSink;
+import org.eclipse.dataspaceconnector.dataplane.spi.result.TransferResult;
+import org.eclipse.dataspaceconnector.dataplane.spi.security.TokenValidationClient;
+import org.eclipse.dataspaceconnector.spi.iam.ClaimToken;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.response.ResponseStatus;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.eclipse.dataspaceconnector.spi.types.domain.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataFlowRequest;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.internalErrors;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.notAuthorizedErrors;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.validationError;
+import static org.eclipse.dataspaceconnector.dataplane.api.common.ResponseFunctions.validationErrors;
+import static org.eclipse.dataspaceconnector.dataplane.core.schema.DataRequestSchema.METHOD;
+import static org.eclipse.dataspaceconnector.dataplane.core.schema.DataRequestSchema.QUERY_PARAMS;
+
+/**
+ * Controller for synchronous data pull APIs of the data plane.
+ */
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
+@Path("/transfer")
+public class DataPlanePublicApiController {
+
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+    private final TokenValidationClient tokenValidationClient;
+    private final DataPlaneManager dataPlaneManager;
+    private final Monitor monitor;
+    private final TypeManager typeManager;
+
+    public DataPlanePublicApiController(TokenValidationClient tokenValidationClient, DataPlaneManager dataPlaneManager, Monitor monitor, TypeManager typeManager) {
+        this.tokenValidationClient = tokenValidationClient;
+        this.dataPlaneManager = dataPlaneManager;
+        this.monitor = monitor;
+        this.typeManager = typeManager;
+    }
+
+    @GET
+    @Path("validate/{token}")
+    public Response getData(@PathParam("token") String token, @Context UriInfo uriInfo, @Context Request request) {
+        // validate and decode input token
+        var tokenValidationResult = tokenValidationClient.validate(token);
+        if (tokenValidationResult.failed()) {
+            return notAuthorizedErrors(tokenValidationResult.getFailureMessages());
+        }
+
+        // create request and perform validation
+        var dataFlowRequest = createDataFlowRequest(tokenValidationResult.getContent(), uriInfo.getQueryParameters(), request.getMethod());
+        var validationResult = dataPlaneManager.validate(dataFlowRequest);
```

Comment:
as we already call the control plane: would it also be an option to retrieve more details like policies we have to fulfill? e.g. "log every request"
And if so: could we remove those details from the token?

Reply:
@MoritzKeppler Yes, we need to do this, probably M3 b/c it needs to be thought through a bit more.